sound/soc/sof/pcm.c

```diff
@@ -346,6 +346,16 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_RELEASE;
 		break;
 	case SNDRV_PCM_TRIGGER_RESUME:
+		if (spcm->stream[substream->stream].suspend_ignored) {
```

Comment:
Fix the commit message

ASoC: SOF: pcm: ignore suspend/resume for D0ix compatible streams

During an system suspend, the PM framework will freeze all applications and the ALSA/ASoC core will suspend all RUNNING PCM streams.

D0ix-compatible PCM streams should not be suspended however. The TRIGGER_SUSPEND event is trapped in such cases, and likewise the TRIGGER_RESUME/START events do not affect the PCM stream state.

In cases where a DAPM widget is connected to those PCM streams, we also ignore PRE_PMU and POST_PMD events to keep the pipelines active.

Reply:
@plbossart @keyonjie I realize there's a bit of a confusion here. There are 2 pipelines in the case of WoV. The one with the PCM and the other one for keyword detection. In our case, we need to keep both the pipelines active, isnt it?
The host pipeline is the one that gets the trigger and the keyword pipeline is handled via DAPM events. So I think we need to say something like this:
"
During system suspend, the PM framework will freeze all applications and the ALSA/ASoC core will suspend all RUNNING PCM streams.

However, D0ix-compatible PCM streams should keep the related pipelines active in the DSP when the system is entering S0ix. The TRIGGER_SUSPEND event is trapped in such cases to prevent the pipelines from being stopped. Likewise, the TRIGGER_RESUME/START events should not affect the pipeline state.

The SOF driver also triggers some pipelines based on the DAPM widgets power events. In such cases,
we also ignore PRE_PMU and POST_PMD events to keep the pipelines active.
"

Reply:
this is not accurate "The TRIGGER_SUSPEND event is trapped in such cases to prevent the pipelines from being stopped."

'stopped' is not correct, it's a change that we don't want.

Should be "The TRIGGER_SUSPEND event is trapped in such cases to prevent the pipeline states from being changed"

Reply:
@plbossart what we did is sending trigger stop to FW at handling this TRIGGER_SUSPEND event, but well, it is also a kind of state change.